### PR TITLE
add libfacedetection

### PR DESCRIPTION
**Pull request by SimonDanisch**
...untested, since I don't have the yggdrasil setup atm! Hope it still works :)

### CODE REVIEW

File: L/libfacedetection/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using BinaryBuilder, Pkg
+
+name = "libfacedetection"
+# I guess it's v3 https://github.com/ShiqiYu/libfacedetection/blob/master/ChangeLog#L3
+# but doesn't really have any versions
+version = v"3"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/ShiqiYu/libfacedetection", "36783e346c596968f6648bdfdc76efc8c6f082a9"),
```
Comment (giordano) [inline L/libfacedetection/build_tarballs.jl:12]:
```suggestion
    GitSource("https://github.com/ShiqiYu/libfacedetection.git", "36783e346c596968f6648bdfdc76efc8c6f082a9"),
```

File: L/libfacedetection/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/libfacedetection
+cmake . -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release -DDEMO=OFF
+cmake --build . --config Release
+cmake --build . --config Release --target install
```
Comment (giordano) [inline L/libfacedetection/build_tarballs.jl:20]:
```suggestion
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
	-DBUILD_SHARED_LIBS=ON \
	-DDEMO=OFF \
	..
make -j${nproc}
make install
```

File: L/libfacedetection/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_SHARED_LIBS=ON \
+	-DDEMO=OFF \
+	..
+make -j${nproc}
+make install
+install_license LICENSE
```
Comment (giordano) [inline L/libfacedetection/build_tarballs.jl:27]:
```suggestion
install_license ../LICENSE
```

File: L/libfacedetection/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/libfacedetection
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_SHARED_LIBS=ON \
+	-DDEMO=OFF \
```
Comment (giordano) [inline L/libfacedetection/build_tarballs.jl:23]:
```suggestion
	-DENABLE_AVX2=OFF \
	-DDEMO=OFF \
```

**Comment by giordano** [comment]
This is another good candidate for the multi-microarchitectures builds

**Comment by SimonDanisch** [comment]
Thanks a lot :) 

>This is another good candidate for the multi-microarchitectures builds

Would that allow turning on avx2?

**Comment by giordano** [comment]
> Would that allow turning on avx2?

Yes, and avx512 on the platforms that allow it

**Comment by SimonDanisch** [comment]
Is this a planned feature, or can we "just" turn this on?

**Comment by giordano** [comment]
Theoretically it's already there, but not tested at all.  And it requires Julia v1.6

**Pull request merged**